plugins/lookup/bitwarden.py

```diff
@@ -0,0 +1,115 @@
+      - be logged into bitwarden
+    short_description: Retrieve secrets from Bitwarden.
+    version_added: 5.4.0
+    description:
+      - Retrieve secrets from Bitwarden.
+    options:
+      _terms:
+        description: Key(s) to fetch values for from login info.
+        required: True
+      field:
+        description: Field to fetch; leave unset to fetch whole response.
+        default: None
+"""
+
+EXAMPLES = """
+- name: "Get 'password' from Bitwarden record named 'a_test'"
+  ansible.builtin.debug:
+    msg: >-
+      {{ lookup('community.general.bitwarden', 'a_test', field='password') }}
+
+- name: "Get full Bitwarden record named 'a_test'"
+  ansible.builtin.debug:
+    msg: >-
+      {{ lookup('community.general.bitwarden', 'a_test') }}
+"""
+
+RETURN = """
+  _raw:
+    description: List of requested field or JSON object of list of matches.
+    type: list
+"""
+
+from subprocess import Popen, PIPE
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.common.text.converters import to_bytes, to_text
+from ansible.parsing.ajson import AnsibleJSONDecoder
+from ansible.plugins.lookup import LookupBase
+
+
+class BitwardenException(AnsibleError):
+    pass
+
+
+class Bitwarden(object):
+
+    def __init__(self, path='bw'):
+        self._cli_path = path
+
+    @property
+    def cli_path(self):
+        return self._cli_path
+
+    @property
+    def logged_in(self):
+        out, err = self._run(['status'], stdin="")
+        decoded = AnsibleJSONDecoder().raw_decode(out)[0]
+        return decoded['status'] == 'unlocked'
+
+    def _run(self, args, stdin=None, expected_rc=0):
+        p = Popen([self.cli_path] + args, stdout=PIPE, stderr=PIPE, stdin=PIPE)
+        out, err = p.communicate(to_bytes(stdin))
+        rc = p.wait()
+        if rc != expected_rc:
+            raise BitwardenException(err)
+        return to_text(out, errors='surrogate_or_strict'), to_text(err, errors='surrogate_or_strict')
+
+    def _get_matches(self, search_value, search_field="name"):
+        """Return matching records whose search_field is equal to key.
+        """
+        out, err = self._run(['list', 'items', '--search', search_value])
+
+        # This includes things that matched in different fields.
+        initial_matches = AnsibleJSONDecoder().raw_decode(out)[0]
+
+        # Filter to only include results from the right field.
+        return list(filter(lambda item: item[search_field] == search_value, initial_matches))
+
+    def get_field(self, field, search_value, search_field="name"):
+        """Return a list of the specified field for records whose search_field match search_value.
+
+        If field is None, return the whole record for each match.
+        """
+        matches = self._get_matches(search_value)
+
+        if field:
+            return [match['login'][field] for match in matches]
+
+        return matches
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
```

Comment:
Why didn't you just use
```suggestion
    def run(self, terms, variables=None, field=None):
```
?

Reply:
I didn't really think about it; I just (ab)used the LastPass module and figured that was the correct signature 😅.